include/swift/AST/Module.h

```diff
@@ -831,13 +831,13 @@ class SourceFile final : public FileUnit {
   std::vector<Decl*> Decls;
 
   /// The list of local type declarations in the source file.
-  TinyPtrVector<TypeDecl*> LocalTypeDecls;
+  llvm::SetVector<TypeDecl *> LocalTypeDecls;
 
   /// A set of special declaration attributes which require the
   /// Foundation module to be imported to work. If the foundation
   /// module is still not imported by the time type checking is
   /// complete, we diagnose.
-  std::map<DeclAttrKind, const DeclAttribute *> AttrsRequiringFoundation;
+  SmallPtrSet<const DeclAttribute *, 4> AttrsRequiringFoundation;
```

Comment:
Technically, this change will make the ordering of diagnostics based on this set non-deterministic.